packages/runtime/src/module/index.ts

```diff
@@ -0,0 +1,132 @@
+import { getFMId, safeToString } from '../utils';
+import { assert } from '../utils/logger';
+import { getRemoteEntry } from '../utils/load';
+import { FederationHost } from '../core';
+import { Global } from '../global';
+import {
+  RemoteEntryExports,
+  Options,
+  Remote,
+  ShareInfos,
+  RemoteInfo,
+} from '../type';
+import { composeKeyWithSeparator } from '@module-federation/sdk';
+
+export type ModuleOptions = ConstructorParameters<typeof Module>[0];
+
+type HostInfo = Remote;
+
+class Module {
+  hostInfo: HostInfo;
+  remoteInfo: RemoteInfo;
+  inited = false;
+  shared: ShareInfos = {};
+  remoteEntryExports?: RemoteEntryExports;
+  lib: RemoteEntryExports | undefined = undefined;
+  loaderHook: FederationHost['loaderHook'];
+  // loading: Record<string, undefined | Promise<RemoteEntryExports | void>> = {};
+
+  constructor({
+    hostInfo,
+    remoteInfo,
+    shared,
+    loaderHook,
+  }: {
+    hostInfo: HostInfo;
+    remoteInfo: RemoteInfo;
+    shared: ShareInfos;
+    plugins: Options['plugins'];
+    loaderHook: FederationHost['loaderHook'];
+  }) {
+    this.hostInfo = hostInfo;
+    this.remoteInfo = remoteInfo;
+    this.shared = shared;
+    this.loaderHook = loaderHook;
+  }
+
+  async getEntry(): Promise<RemoteEntryExports> {
+    if (this.remoteEntryExports) {
+      return this.remoteEntryExports;
+    }
+
+    // Get remoteEntry.js
+    const remoteEntryExports = await getRemoteEntry({
+      remoteInfo: this.remoteInfo,
+      remoteEntryExports: this.remoteEntryExports,
+      createScriptHook: (url: string) => {
+        const res = this.loaderHook.lifecycle.createScript.emit({ url });
+        if (res instanceof HTMLScriptElement) {
+          return res;
+        }
+        return;
+      },
+    });
+    assert(
+      remoteEntryExports,
+      `remoteEntryExports is undefined \n ${safeToString(this.remoteInfo)}`,
+    );
+
+    this.remoteEntryExports = remoteEntryExports;
+    return this.remoteEntryExports;
+  }
+
+  // eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types
+  async get(expose: string, options?: { loadFactory?: boolean }) {
+    const { loadFactory = true } = options || { loadFactory: true };
+
+    // Get remoteEntry.js
+    const remoteEntryExports = await this.getEntry();
+
+    if (!this.inited) {
+      const globalShareScope = Global.__FEDERATION__.__SHARE__;
+      const remoteShareScope = this.remoteInfo.shareScope || 'default';
+
+      if (!globalShareScope[remoteShareScope]) {
+        globalShareScope[remoteShareScope] = {};
+      }
+      const shareScope = globalShareScope[remoteShareScope];
+
+      // TODO: compat logic , it could be moved after providing startup hooks
```

Comment:
Vmok remoteEntry init parameters are not same as standard mf , so i have to compat it temporary.

We can add startup hooks which is 1.5 feature to remove this code completely. @ScriptedAlchemy

Reply:
Yep thats fine, any compat stuff we need is no problem. 
Great to see we have a existing use case where startup hooks would help us for VMOK